babel-plugin-transform-jsx/README.md

```diff
@@ -304,6 +305,14 @@ function MyComponent() {
 
 In the transformed object instead of having the string `foo.bar` for `elementName`, it would instead reference `foo.bar` the object property.
 
+However, if you want to use component element names as shown in the React example above, you can set the `useVariables` option to enable support:
+
+```json
+{
+  "plugins": [["transform-jsx", { "useVariables": true }]]
```

Comment:
The option name was bothering me as well. What do you think of `allowComponentElements` to be more specific about the what the feature actually is? In addition, it would be possible to use global variables if we opt for your approach of pattern-matching component element names.